test-requirements.txt

```diff
@@ -4,7 +4,8 @@
 #
 #    pip-compile --output-file=test-requirements.txt setup.py test-requirements.in
 #
-astroid==2.5.8
+astroid==3.1.0; python_version >= "3.8"
+astroid<3.1.0; python_version < "3.8"
```

Comment:
Are these hand-edits?  The `.txt` files should be generated from `.in` files (see `Makefile`).   But dealing with dependencies that change by Python version is difficult, pip-tools doesn't support it too well.

It may need a complication like maintaining a separate `test-requirements.txt` file for Python 3.7.  For example, copy current `test-requirements.txt` to `test-requirements_py3.7.txt`.

(Or drop Python 3.7, which would be unfortunate.)

> astroid<3.1.0

Pinned requirement file should have `==` only.

Reply:
Thank you for your reply. I ended up making hand-edits, so I've redone it. https://github.com/groove-x/trio-util/pull/25/commits/b8c03bdac3666b4623df12b96416f84dd9eb6bf9

> (Or drop Python 3.7, which would be unfortunate.)

Sadly, I'm considering dropping Python 3.7. What are your thoughts? I'm also willing to create a separate test-requirements_py3.7.txt if needed.

Reply:
I don't want to drop Python 3.7, since it's still an active version, and there isn't a technical reason to drop it.

Popular packages like numpy have dropped 3.7, which causes a lot of problems for package users, and I'd prefer not to follow it.